File: leetcode/easy/two-sum.py
#!/usr/bin/env python

def twoSum(nums, target):
    mhash = {}
    for i in range(len(nums)):
        diff = target - nums[i]
        if diff in mhash.keys():
            return [mhash.get(diff), i]
        mhash[nums[i]] = i
    return []

nums = [2, 7 , 11, 15]
print(twoSum(nums, 9))

File: leetcode/easy/two-sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    mhash := make(map[int]int)

    for i := 0; i < len(nums); i++ {
        diff := target - nums[i]

        if _, ok := mhash[diff]; ok {
            return []int{mhash[diff], i}
        }

        mhash[nums[i]] = i
    }
    return []int{}
}

func main() {
    a := twoSum([]int{2, 7, 11, 15}, 9)
    fmt.Println(a)
}

